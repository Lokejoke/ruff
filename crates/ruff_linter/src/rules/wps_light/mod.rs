//! Rules from [wps-light](https://pypi.org/project/wps-light/).
pub(crate) mod rules;

#[cfg(test)]
mod tests {
    use std::convert::AsRef;
    use std::path::Path;

    use anyhow::Result;
    use test_case::test_case;

    use crate::registry::Rule;
    use crate::test::test_path;
    use crate::{assert_messages, settings};

    #[test_case(Rule::ListMultiplication, Path::new("WPS435.py"))]
    #[test_case(Rule::ReservedArgumentAsVariable, Path::new("WPS117.py"))]
    #[test_case(Rule::UnderscoresInNumber, Path::new("WPS303.py"))]
    #[test_case(Rule::AssignmentToSubscriptSlice, Path::new("WPS362.py"))]
    fn rules(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.as_ref(), path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("wps_light").join(path).as_path(),
            &settings::LinterSettings::for_rule(rule_code),
        )?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }
}
