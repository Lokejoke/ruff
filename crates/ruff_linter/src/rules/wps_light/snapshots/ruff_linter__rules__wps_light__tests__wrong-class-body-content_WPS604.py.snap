---
source: crates/ruff_linter/src/rules/wps_light/mod.rs
snapshot_kind: text
---
WPS604.py:87:5: WPS604 Wrong statement inside class definition
   |
86 |       # Conditional logic inside the class (not recommended but valid)
87 |       if math.pi > 3:
   |  _____^
88 | |         def conditionally_defined_method(self):
89 | |             print("This method exists because math.pi > 3.")
   | |____________________________________________________________^ WPS604
90 |       
91 |       # For loop inside the class (unusual but valid)
   |

WPS604.py:92:5: WPS604 Wrong statement inside class definition
   |
91 |       # For loop inside the class (unusual but valid)
92 |       for i in range(3):
   |  _____^
93 | |         exec(f"def loop_method_{i}(self): print('Loop-defined method {i}')")
   | |____________________________________________________________________________^ WPS604
94 |       
95 |       # Docstrings for the class
   |
